src/site/content/en/blog/codelab-sign-up-form-best-practices/index.md

```diff
@@ -0,0 +1,269 @@
+range of users on mobile and desktop.
+
+In this step you'll add CSS to make the form easier to use.
+
+Copy and paste all the following CSS into your own `style.css` file:
+
+{% Glitch {
+  id: 'sign-up-form-codelab-2',
+  path: 'css/main.css',
+  height: 400,
+  previewSize: 0
+} %}
+
+Click **View App** to see your styled sign-up form. Then click **View Source** to return to 
+`style.css`.
+
+* Does this CSS work for a variety of browsers and screen sizes?
+
+* Try adjusting `padding`, `margin` and `font-size` to suit your test devices.
+
+* The CSS is mobile-first. [Media queries](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Media_queries) 
+are used to apply CSS rules for viewports that are at least `400px` wide, and then again for 
+viewports that are at least `500px` wide. Are these [breakpoints](https://www.browserstack.com/guide/responsive-design-breakpoints) 
+adequate? How should you choose breakpoints for forms?
+
+{% Aside %}
+**Viewport** in this context means the display area available for your page. Mobile phones have a 
+smaller viewport than laptops, and a small browser window on a desktop monitor has a smaller 
+viewport than a maximized browser window.
+{% endAside %}
+
+
+## Step 3: Add attributes to enable built-in browser features
+
+In this step you add attributes to your input elements to enable the browser to store and autofill 
+form field values, and warn of fields with missing or invalid data.
+
+Update your `index.html` file so the form code looks like this:
+
+```html
+<form action="#" method="post">
+        
+  <h1>Sign up</h1>
+  
+  <section>        
+    <label for="name">Full name</label>
+    <input id="name" name="name" autocomplete="name" 
+           pattern="[\p{L}\.\- ]+" required>
+  </section>
+
+  <section>        
+    <label for="email">Email</label>
+    <input id="email" name="email" autocomplete="username"
+           type="email" required>
+  </section>
+  
+  <section>
+    <label for="password">Password</label>
+    <input id="password" name="password" autocomplete="new-password" 
+           type="password" minlength="8" required>
+  </section>
+  
+  <button id="sign-up">Sign up</button>
+  
+</form>
+```
+
+The `type` values do a lot:
+* `type="password"` obscures text as it's entered and enables the browser's 
+[password manager](https://passwords.google.com/) to suggest a strong password. 
+* `type="email"` provides basic validation and ensures mobile users get an appropriate keyboard.
+
+Try submitting the form with an empty field. The browser won't submit the form, and it prompts to 
+complete missing data and sets focus. That's because you added the `require` attribute to all the 
+inputs. Now try submitting with a password that has less than eight characters. The browser warns that 
+the password isn't long enough and sets focus on the password input. The same works for `pattern` 
```

Comment:
```suggestion
inputs. Now try submitting with a password that has less than eight characters. The browser warns that 
the password isn't long enough and sets focus on the password input because of the `minlength="8"` attribute. The same works for `pattern` 
```